lib/view_component/system_test_helpers.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module ViewComponent
+  module SystemTestHelpers
+    include TestHelpers
+    include Capybara::DSL
+
+    def page
+      Capybara.current_session
+    end
+
+    def visit_rendered_component_in_browser(component, **options)
+      layout = options[:layout] || false
+
+      opts = {
+        layout: layout,
+        locals: {
+          render_args: {
+            component: component,
+            template: "view_components/preview",
+            hide_preview_source: true
+          }
+        }
+      }
+      html = controller.render_to_string("view_components/preview", opts)
+
+      # Add './tmp/view_component_integrations/' directory if it doesn't exist to store the rendered component html
+      FileUtils.mkdir_p("./tmp/view_component_integrations/") unless Dir.exist?("./tmp/view_component_integrations/")
+
+      # Write to temporary file to contain fully rendered component
+      # within a browser
+      file = Tempfile.new(["rendered_#{component.class.name}", ".html"], "tmp/view_component_integrations")
+      file.write(html)
+      file.rewind
+
+      filename = file.path.split("/").last
+
+      # Visit the file that contains the HTML
+      visit "/view_components_system_test_entrypoint?file=#{filename}"
+
+      # Erase temporary file
+      file.unlink
```

Comment:
One downside to moving way from handling `visit` is that it would be trickier to handle erasing temporary files. Perhaps we could yield the entrypoint instead? Allowing for something like this:

```ruby
with_rendered_component_path(MyComponent.new, layout: "special") do |path|
  visit(path)
end
```

Reply:
I like this idea and the `file.unlink` could be in the ensure block so we permanently delete it after it's been used. Let me try that out.